### ISSUE TITLE
Deal with lint issues from flutter analyze of at_client_mobile

### ISSUE THREAD

**Issue by cpswan** [atsign-foundation/at_client_sdk#309]
**Describe the bug**
Since switching to package:lints/recommended.yaml CI is failing due to a large number of lint issues found:

```
  flutter analyze
  shell: /usr/bin/bash -e {0}
  env:
    PUB_CACHE: /home/runner/.pub-cache
Analyzing at_client_mobile...                                   

   info • Don't import implementation files from another package • lib/src/at_client_auth.dart:1:8 • implementation_imports
   info • Prefer typing uninitialized variables and fields • lib/src/at_client_auth.dart:64:11 • prefer_typing_uninitialized_variables
   info • Name non-constant identifiers using lowerCamelCase • lib/src/at_client_auth.dart:86:9 • non_constant_identifier_names
   info • Name non-constant identifiers using lowerCamelCase • lib/src/at_client_auth.dart:114:13 • non_constant_identifier_names
   info • Don't import implementation files from another package • lib/src/at_client_service.dart:6:8 • implementation_imports
   info • Prefer using lowerCamelCase for constant names • lib/src/at_client_service.dart:312:23 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/at_client_service.dart:313:23 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/at_client_service.dart:314:23 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/at_client_service.dart:315:23 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/at_client_service.dart:316:23 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/auth_constants.dart:1:14 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/auth_constants.dart:2:14 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/auth_constants.dart:3:14 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/auth_constants.dart:4:14 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/auth_constants.dart:5:14 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/auth_constants.dart:6:14 • constant_identifier_names
   info • Prefer typing uninitialized variables and fields • lib/src/keychain_manager.dart:54:9 • prefer_typing_uninitialized_variables
   info • Provide a deprecation message, via @Deprecated("message") • lib/src/keychain_manager.dart:66:3 • provide_deprecation_message
   info • Prefer typing uninitialized variables and fields • lib/src/keychain_manager.dart:68:9 • prefer_typing_uninitialized_variables
   info • Provide a deprecation message, via @Deprecated("message") • lib/src/keychain_manager.dart:80:3 • provide_deprecation_message
   info • Prefer typing uninitialized variables and fields • lib/src/keychain_manager.dart:82:9 • prefer_typing_uninitialized_variables
   info • Prefer typing uninitialized variables and fields • lib/src/keychain_manager.dart:94:9 • prefer_typing_uninitialized_variables
   info • Prefer using lowerCamelCase for constant names • lib/src/onboarding_status.dart:2:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/onboarding_status.dart:3:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/onboarding_status.dart:4:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/onboarding_status.dart:5:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/onboarding_status.dart:6:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/onboarding_status.dart:7:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/onboarding_status.dart:8:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/onboarding_status.dart:9:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/onboarding_status.dart:10:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/onboarding_status.dart:11:3 • constant_identifier_names
   info • Prefer using lowerCamelCase for constant names • lib/src/onboarding_status.dart:12:3 • constant_identifier_names

33 issues found. (ran in 6.9s)
Error: Process completed with exit code 1.
```

**To Reproduce**
Take a look at any recent [workflow runs for at_client_sdk CI](https://github.com/atsign-foundation/at_client_sdk/actions/workflows/at_client_sdk.yaml)

**Expected behavior**
No lint issues

**Comment by adityathakurxd** [comment]
Hello,
Should the fix of this issue be made by updating the `analysis_options.yaml` something like:
```
errors:
        avoid_print: ignore
        constant_identifier_names: ignore
        prefer_typing_uninitialized_variables: ignore
```
or, to work on this do we need to update the variable names to lowerCamelCase and make other changes?

I believe updating the `analysis_options.yaml` would be a temporary fix for CI and the changes to variable names and others can be made gradually by removing ignore from the `analysis_options.yaml`.

**Comment by adityathakurxd** [comment]
Also, the package:lints can be migrated to package:flutter_lints (as suggested by Flutter team [here](https://flutter.dev/docs/release/breaking-changes/flutter-lints-package)).

**Comment by yahu1031** [comment]
I will push the changes tonight

**Comment by yahu1031** [comment]
@adityathakurxd correct, But you see we don't really use any flutter widgets in the package. so we don't necessarily need that to be `flutter_lints`

**Comment by yahu1031** [comment]
And also there is a fix for this in PR #330

**Comment by adityathakurxd** [comment]
Hey @yahu1031,
Thank you for letting me know! I had totally forgotten about this issue. Got all these notifs just now (2 months later).

I had I remember worked on it but hadn't opened a Pull Request just in case someone from the team had a different and better approach. 

The mentioned PR closing this issue doesn't specify much in the description. Seems like a bad open source practice to me.

But, Thank you anyway!
Possibly in the future can work on an unassigned issue.
